nixos/modules/services/networking/wireguard.nix

```diff
@@ -153,35 +182,43 @@ let
   wgCommand = "${pkgs.wireguard}/bin/wg";
 
   generateUnit = name: values:
+    assert (values.privateKey != null || values.privateKeyFile != null);
```

Comment:
Erm, right.

You should error out if both privateKey and privateKeyFile are set, or if both presharedKey and presharedKeyFile are set.